qiskit/pulse/macros.py

```diff
@@ -104,3 +232,33 @@ def measure_all(backend) -> Schedule:
         A schedule corresponding to the inputs provided.
     """
     return measure(qubits=list(range(backend.configuration().n_qubits)), backend=backend)
+
+
+def schedule_remapping_memory_slot(schedule: Schedule, qubit_mem_slots: Dict[int, int]) -> Schedule:
```

Comment:
@TsafrirA are you happy with this API? This is public so we cannot easily change after release. If you have any concern we can turn this into protected.

Reply:
I changed at https://github.com/Qiskit/qiskit-terra/pull/9987/commits/b075cb394e0fed1df63ec1aad5a24b5976fc1c3a.